src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs

```diff
@@ -69,32 +69,20 @@ public IMoBiCommand SetContainerMode(IBuildingBlock buildingBlock, IContainer co
          return new SetContainerModeCommand(buildingBlock, container, containerMode).Run(_context);
       }
 
-
-      //TODO EXACT COPY PASTE from InteractionTasksForContainerBase
-      private IReadOnlyList<NeighborhoodBuilder> getConnectingNeighborhoods(IEnumerable<IContainer> existingItems, ISpatialStructure tmpSpatialStructure)
+      protected override IMoBiCommand GetRenameCommandFor(IContainer container, IBuildingBlock buildingBlock, string newName, string objectType)
       {
-         var allImportedContainers = existingItems.SelectMany(
-            cont => cont.GetAllContainersAndSelf<IContainer>().Where(x => !x.IsAnImplementationOf<IParameter>())).ToList();
-
-         var neighborhoods = new List<NeighborhoodBuilder>();
-         foreach (var neighborhood in tmpSpatialStructure.Neighborhoods)
-         {
-            var firstFound = false;
-            var secondFound = false;
-            foreach (var cont in allImportedContainers)
-            {
-               var contObjectPath = _objectPathFactory.CreateAbsoluteObjectPath(cont);
-               if (Equals(neighborhood.FirstNeighborPath, contObjectPath))
-                  firstFound = true;
-
-               if (Equals(neighborhood.SecondNeighborPath, contObjectPath))
-                  secondFound = true;
-            }
+         //when renaming a container in a spatial structure, we need to ensure that we are also renaming the path in the neighborhood
+         if (buildingBlock.IsAnImplementationOf<ISpatialStructure>() && containerCanBePartOfNeighborhoodPath(container))
+            return new RenameContainerCommand(container, newName, buildingBlock.DowncastTo<ISpatialStructure>());
 
-            if (firstFound && secondFound) neighborhoods.Add(neighborhood);
-         }
+         return base.GetRenameCommandFor(container, buildingBlock, newName, objectType);
+      }
 
-         return neighborhoods;
+      private bool containerCanBePartOfNeighborhoodPath(IContainer container)
```

Comment:
@Yuri05 I am not sure when I should use the special command. I said for now all container except the one called MoleculeProperties, the parameter (DistributedParameter) and the NeighborhoodBuilderthemselves (also container)

Reply:
Is ok I think.